assets/test/specs/modalDialog.spec.js

```diff
@@ -0,0 +1,233 @@
+    beforeEach(function() {
+      modalDialog().init(); // init modal-dialog module
+    });
+
+    describe('when clicking a [data-modaldialog-action="close"] control in modal-dialog content', function() {
+      var $contentCloseLink;
+
+      beforeEach(function() {
+        $contentCloseLink = $content2.find('a[data-modaldialog-action="close"]');
+        expect($contentCloseLink.length).toBe(1);
+        $contentCloseLink.click();
+      });
+      
+      it('content is not visible', function() {
+        expect($content2.is(':not(:visible)')).toBe(true);
+        expect($content1.is(':not(:visible)')).toBe(true);
+      });
+      
+      it("body becomes scroll-able", function(){
+        expect($body.hasClass('scroll-off')).toBe(false);
+      });
+
+      it("modal-dialog is hidden", function(){
+        expect($modalDialog.hasClass('modal-dialog--hidden')).toBe(true);
+      });
+    });
+
+    describe('when clicking a [data-modaldialog-action="close"] control in document content', function() {
+      var $documentCloseLink;
+
+      beforeEach(function() {
+        $documentCloseLink = $actions.find('[data-modaldialog-action="close"][data-modaldialog-target="content-2"]');
+        expect($documentCloseLink.length).toBe(1);
+        $documentCloseLink.click();
+      });
+
+      it('content is not visible', function() {
+        expect($content2.is(':not(:visible)')).toBe(true);
+        expect($content1.is(':not(:visible)')).toBe(true);
+      });
+
+      it("body becomes scroll-able", function(){
+        expect($body.hasClass('scroll-off')).toBe(false);
+      });
+
+      it("the modal-dialog is hidden", function(){
+        expect($modalDialog.hasClass('modal-dialog--hidden')).toBe(true);
+      });
+    });
+    
+    it('when user uses escape key', function() {
+      $documentOpenLink = $actions.find('[data-modaldialog-action="open"][data-modaldialog-target="content-1"]');
+
+      //var e = $.Event("keyup");
+      //e.which = 27;
+      //$content2.trigger(e);
+      
+      simulateKeyEvent($content2, 'keyup', 27);
+      
+      expect($content2.is(':not(:visible)')).toBe(true);
+      expect($content1.is(':not(:visible)')).toBe(true);
+      expect($body.hasClass('scroll-off')).toBe(false);
+      expect($modalDialog.hasClass('modal-dialog--hidden')).toBe(true);
+    });
+  });
+
+  describe('and the modal-dialog opens', function() {
+    beforeEach(function() {
+      modalDialog().init(); // init modal-dialog module
+    });
+
+    describe('when clicking a [data-modaldialog-action="open"] control in modal-dialog content', function() {
+      beforeEach(function() {
+        $content2.find('a[data-modaldialog-action="close"]').click();
+        expect($content2.is(':not(:visible)')).toBe(true);
+        expect($content1.is(':not(:visible)')).toBe(true);
+        $actions.find('[data-modaldialog-action="open"][data-modaldialog-target="content-1"]').click();
+      });
+
+      it('content is visible', function() {
```

Comment:
it has been mentioned previously when I did some tests ideally to use the word *should* in tests